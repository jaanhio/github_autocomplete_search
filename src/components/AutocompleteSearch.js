import React from 'react';
import styled from 'styled-components';

const SearchbarWrapper = styled.div`
  display: flex;
`;

const StyledButton = styled.button`
  height: 36px;
  display: flex;
  align-items: center;
  width: 60px;
  justify-content: center;
  background-color: #2769CF;
  border: none;
`;

const StyledInput = styled.input`
  width: 60vw;
  font-size: 1em;
  padding-left: 10px;
  max-width: 400px;
  border: 1px solid #CDCDCD;
  font-weight: 200;
`;

const AutocompleteSearch = ({ autocompleteSuggestions, handleSearchChange, searchTerm, getSearchResults }) => {

  const renderOptions = autocompleteSuggestions.map(suggestion => {
    return (
      <option key={suggestion.node_id}>{suggestion.name}</option>
    );
  });

  return (
    <SearchbarWrapper>
      <select name='search-option'>
        <option value='1'>Repository</option>
        <option value='2'>User</option>
      </select>
      <StyledInput list='repoList' onChange={handleSearchChange} value={searchTerm} placeholder='Find Github repo (e.g express)' />
      <datalist id='repoList'>
        {renderOptions}
      </datalist>
      <StyledButton onClick={getSearchResults}>
        <i class="material-icons" style={{ color: 'white' }}>search</i>
      </StyledButton>
    </SearchbarWrapper>
  )
}

export default AutocompleteSearch;