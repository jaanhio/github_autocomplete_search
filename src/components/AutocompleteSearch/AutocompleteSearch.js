import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const SearchbarWrapper = styled.div`
  display: flex;
`;

const StyledButton = styled.button`
  height: 36px;
  display: flex;
  align-items: center;
  width: 60px;
  justify-content: center;
  background-color: #2769CF;
  border: none;
`;

const StyledSelect = styled.select`
  height: 36px;
  font-size: 0.6em;
`;

const StyledInput = styled.input`
  width: 60vw;
  font-size: 1em;
  padding-left: 10px;
  max-width: 400px;
  border: 1px solid #CDCDCD;
  font-weight: 200;
`;

const AutocompleteSearch = ({ autocompleteSuggestions, handleSearchChange, handleSelectChange, searchTerm, searchCategory, getSearchResults }) => {

  const renderOptions = autocompleteSuggestions.map(suggestion => {
    return searchCategory === 'repo' ? (
      <option key={suggestion.node_id}>{suggestion.name}</option>
    ) : (<option key={suggestion.node_id}>{suggestion.login}</option>)
  });

  const placeholderMessage = searchCategory === 'repo' ? 'Find Github repo (e.g express)' : 'Find Github user';

  return (
    <SearchbarWrapper>
      <StyledSelect name='search-option' onChange={handleSelectChange}>
        <option value='repo'>Repository</option>
        <option value='user'>User</option>
      </StyledSelect>
      <StyledInput list='repoList' onChange={handleSearchChange} value={searchTerm} placeholder={placeholderMessage} />
      <datalist id='repoList'>
        {renderOptions}
      </datalist>
      <StyledButton onClick={getSearchResults}>
        <i className="material-icons" style={{ color: 'white' }}>search</i>
      </StyledButton>
    </SearchbarWrapper>
  )
}

AutocompleteSearch.propTypes = {
  autocompleteSuggestions: PropTypes.array.isRequired,
  handleSearchChange: PropTypes.func,
  handleSelectChange: PropTypes.func,
  searchTerm: PropTypes.string.isRequired,
  searchCategory: PropTypes.string.isRequired,
  getSearchResults: PropTypes.func
};

AutocompleteSearch.defaultProps = {
  autocompleteSuggestions: [],
  searchTerm: '',
  searchCategory: 'repo'
};

export default AutocompleteSearch;