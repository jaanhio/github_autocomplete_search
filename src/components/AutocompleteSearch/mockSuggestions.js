const mockRepoSuggestions = [
  {
    "node_id": "12345678",
    "name": "suggestion1",
  },
  {
    "node_id": "87654321",
    "name": "suggestion2",
  },
];

const mockUserSuggestions = [
  {
    "node_id": "12345678",
    "login": "user1",
  },
  {
    "node_id": "87654321",
    "login": "user2",
  }
];

module.exports = {
  mockRepoSuggestions,
  mockUserSuggestions
}