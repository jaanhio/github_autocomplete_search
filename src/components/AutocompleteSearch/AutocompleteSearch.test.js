import React from 'react';
import AutocompleteSearch from './AutocompleteSearch';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';


describe('<AutocompleteSearch /> component', () => {
  it('render without throwing error', () => {
    const wrapper = shallow(<AutocompleteSearch />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
  describe('Search input box placeholder message', () => {
    it('renders default repo placeholder message', () => {
      const wrapper = shallow(<AutocompleteSearch />);
      const searchInputComponent = wrapper.props().children[1];
      expect(searchInputComponent.props.placeholder).toEqual('Find Github repo (e.g express)');
    });
    it('renders user placeholder message if search category is selected', () => {
      const searchCategory = 'user';
      const wrapper = shallow(<AutocompleteSearch searchCategory={searchCategory} />);
      const searchInputComponent = wrapper.props().children[1];
      expect(searchInputComponent.props.placeholder).toEqual('Find Github user');
    });
  });
  describe('Search category select element', () => {
    it('renders 2 options in search category select element ', () => {
      const wrapper = shallow(<AutocompleteSearch />);
      const selectComponent = wrapper.props().children[0];
      expect(selectComponent.props.children.length).toEqual(2);
    });
    it('updates searchCategory state/props on change', () => {
      const wrapper = shallow(<AutocompleteSearch />);
      // console.log(wrapper.props());
    });
  });
  describe('Search button', () => {
    it('gets results on click', );
    const wrapper = mount(<AutocompleteSearch />);
    console.log(wrapper.children());
  });
});