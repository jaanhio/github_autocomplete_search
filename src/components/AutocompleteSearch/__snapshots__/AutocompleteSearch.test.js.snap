// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutocompleteSearch /> component render without throwing error with mock repo suggestions 1`] = `
<styled.div>
  <styled.select
    name="search-option"
  >
    <option
      value="repo"
    >
      Repository
    </option>
    <option
      value="user"
    >
      User
    </option>
  </styled.select>
  <styled.input
    list="repoList"
    placeholder="Find Github repo (e.g express)"
    value=""
  />
  <datalist
    id="repoList"
  >
    <option
      key="12345678"
    >
      suggestion1
    </option>
    <option
      key="87654321"
    >
      suggestion2
    </option>
  </datalist>
  <styled.button>
    <i
      className="material-icons"
      style={
        Object {
          "color": "white",
        }
      }
    >
      search
    </i>
  </styled.button>
</styled.div>
`;

exports[`<AutocompleteSearch /> component render without throwing error with mock user suggestions 1`] = `
<styled.div>
  <styled.select
    name="search-option"
  >
    <option
      value="repo"
    >
      Repository
    </option>
    <option
      value="user"
    >
      User
    </option>
  </styled.select>
  <styled.input
    list="repoList"
    placeholder="Find Github user"
    value=""
  />
  <datalist
    id="repoList"
  >
    <option
      key="12345678"
    >
      user1
    </option>
    <option
      key="87654321"
    >
      user2
    </option>
  </datalist>
  <styled.button>
    <i
      className="material-icons"
      style={
        Object {
          "color": "white",
        }
      }
    >
      search
    </i>
  </styled.button>
</styled.div>
`;

exports[`<AutocompleteSearch /> component render without throwing error without suggestions 1`] = `
<styled.div>
  <styled.select
    name="search-option"
  >
    <option
      value="repo"
    >
      Repository
    </option>
    <option
      value="user"
    >
      User
    </option>
  </styled.select>
  <styled.input
    list="repoList"
    placeholder="Find Github repo (e.g express)"
    value=""
  />
  <datalist
    id="repoList"
  />
  <styled.button>
    <i
      className="material-icons"
      style={
        Object {
          "color": "white",
        }
      }
    >
      search
    </i>
  </styled.button>
</styled.div>
`;
